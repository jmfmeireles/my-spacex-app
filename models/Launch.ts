export interface LaunchBasicData {
  id: number;
  details: string | null;
  launch_date_local: string;
  mission_name: string;
}

export interface LaunchDetailedData extends LaunchBasicData {
  links: MissionLinks;
  ships: Ship[];
  rocket: Rocket;
  launch_site: LaunchSite;
}

export interface MissionLinks {
  video_link: string | null;
  article_link: string | null;
  mission_patch: string | null;
  flickr_images: string[];
}

export interface Ship {
  name: string;
  type: string;
  url: string;
}

export interface Rocket {
  rocket_name: string;
}

export interface LaunchSite {
  site_name: string;
}
