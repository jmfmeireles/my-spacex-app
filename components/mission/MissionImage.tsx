import { useContext } from 'react';
import Image from 'next/image';

import styles from '../../styles/Mission.module.css';
import { MissionLinks } from '../../models/Launch';
import TranslationContext from '../../context/Translation';

const MissionImage = (props: { missionLinks: MissionLinks }) => {
  const { t } = useContext(TranslationContext);
  const imageSrc: string | null = props.missionLinks.flickr_images.length
    ? props.missionLinks.flickr_images[0]
    : props.missionLinks.mission_patch;
  return (
    <>
      {imageSrc && (
        <div className={styles.image}>
          <Image
            src={imageSrc}
            alt={t.missionImage}
            layout="responsive"
            width="100%"
            height="100%"
            priority={true}
          />
        </div>
      )}
    </>
  );
};

export default MissionImage;
