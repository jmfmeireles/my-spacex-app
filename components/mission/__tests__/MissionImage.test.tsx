import { render } from '@testing-library/react';
import { MissionLinks } from '../../../models/Launch';
import MissionImage from '../MissionImage';

const defaultMissionLinksUsedForTest: MissionLinks = {
  video_link: 'https://youtu.be/aVFPzTDCihQ',
  article_link:
    'https://spaceflightnow.com/2020/11/21/international-satellite-launches-to-extend-measurements-of-sea-level-rise/',
  mission_patch: null,
  flickr_images: [
    'https://live.staticflickr.com/65535/50630802488_8cc373728e_o.jpg',
    'https://live.staticflickr.com/65535/50631642722_3af8131c6f_o.jpg',
  ],
};

describe('Mission Image', () => {
  it('renders first image of flickr_images properly', () => {
    expect(
      getRenderedComponent(defaultMissionLinksUsedForTest),
    ).toMatchSnapshot();
  });

  it('renders mission patch properly', () => {
    expect(
      getRenderedComponent({
        ...defaultMissionLinksUsedForTest,
        flickr_images: [],
        mission_patch:
          'https://live.staticflickr.com/65535/50630802488_8cc373728e_o.jpg',
      }),
    ).toMatchSnapshot();
  });

  it('does not render anything', () => {
    expect(
      getRenderedComponent({
        ...defaultMissionLinksUsedForTest,
        flickr_images: [],
      }),
    ).toMatchSnapshot();
  });

  function getRenderedComponent(missionLinks: MissionLinks) {
    return render(<MissionImage missionLinks={missionLinks} />).container;
  }
});
