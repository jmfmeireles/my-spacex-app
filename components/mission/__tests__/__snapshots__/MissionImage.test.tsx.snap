// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Image does not render anything 1`] = `<div />`;

exports[`Mission Image renders first image of flickr_images properly 1`] = `
<div>
  <div
    class="image"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
      />
      <img
        alt="Image relative to the mission"
        data-nimg="responsive"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </span>
  </div>
</div>
`;

exports[`Mission Image renders mission patch properly 1`] = `
<div>
  <div
    class="image"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
      />
      <img
        alt="Image relative to the mission"
        data-nimg="responsive"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Flive.staticflickr.com%2F65535%2F50630802488_8cc373728e_o.jpg&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </span>
  </div>
</div>
`;
